common/web/gesture-recognizer/src/engine/touchEventEngine.ts

```diff
@@ -185,13 +232,34 @@ export class TouchEventEngine<HoveredItemType, StateToken = any> extends InputEv
       }
     }
 
+    /*
+      Using the Promise map built in touchStart, we can retrieve a Promise for the source linked
+      to this event and closure-capture it for the closure queued below.
+    */
+    const capturedSourcePromises = new Map<number, Promise<GestureSource<ItemType, StateToken>>>();
+    for(let i = 0; i < event.touches.length; i++) {
+      const touchId = event.touches.item(i).identifier;
+      capturedSourcePromises.set(touchId, this.pendingSourcePromises.get(touchId).corePromise);
+    }
```

Comment:
The big difference from #10843 is that we can closure-capture the Promises at the original time of the event, unlike the `GestureSource` instances.  #10843 tried to make the mapping work asynchronously, but there were cases that arose in which the former, more loosely-bound style simply couldn't work consistently.

That was likely because the mapping could see an entry erased by a different `GestureSource` (conceptually) that was reusing the same identifier.